streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -908,20 +908,25 @@ private void initializeAndRestorePhase() {
 
     // Check if the topology has been updated since we last checked, ie via #addNamedTopology or #removeNamedTopology
     private void checkForTopologyUpdates() {
-        if (lastSeenTopologyVersion < topologyMetadata.topologyVersion() || topologyMetadata.isEmpty()) {
-            lastSeenTopologyVersion = topologyMetadata.topologyVersion();
-            taskManager.handleTopologyUpdates();
-
```

Comment:
For line 915 below: do we still need this line? Could we just inline the `Condition topologyCV` waiting within this while loop now?

Reply:
I think we might be able to now

Reply:
I ended up not doing this to avoid exposing locks without need. We did get rid of one of the loops so I hope that helps